credential/template/dataset.go

```diff
@@ -0,0 +1,127 @@
+package template
+
+import (
+	"bytes"
+	_ "embed"
+	"errors"
+	gotemplate "text/template"
+	"time"
+
+	"github.com/axone-protocol/axone-sdk/credential"
+	"github.com/axone-protocol/axone-sdk/dataverse"
+	"github.com/google/uuid"
+)
+
+//go:embed vc-desc-tpl.jsonld
+var datasetTemplate string
+
+var _ credential.Descriptor = NewDataset()
```

Comment:
```suggestion
var _ credential.Descriptor DatasetDescriptor{}
```

This require to implement the `Descriptor` methods on the instance directly, I think this is ok but I'll let it to you :)